### PR TITLE
shim: Add trace config option

### LINKED ISSUE #934: Add trace option for shim
Add a new config option to enable shim tracing (see https://github.com/kata-containers/shim/issues/123).

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a new `enable_tracing` option to `configuration.toml` to enable shim tracing.
    
Fixes #934.
    
Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Blocked on:

- https://github.com/kata-containers/shim/issues/123
- https://github.com/kata-containers/runtime/pull/937 (this PR includes the same fix - will rebase when that lands...)

**Comment by jodh-intel** [comment]
Removing dnm label now https://github.com/kata-containers/shim/pull/124 is merged...

**Comment by jodh-intel** [comment]
Centos CI failing due to #944.

**Comment by jodh-intel** [comment]
Umm, I mean https://github.com/kata-containers/tests/pull/965 :)

File: virtcontainers/kata_shim.go [hunk 1]
```diff
@@ -57,5 +59,17 @@ func (s *kataShim) start(sandbox *Sandbox, params ShimParams) (int, error) {
 		args = append(args, "-log", "debug")
 	}
 
+	if config.Trace {
+		args = append(args, "-trace")
+
+		// FIXME: CRITICAL: should use "DisableNewNetNs" instead
```
Comment (devimc) [inline virtcontainers/kata_shim.go]: please file an issue :bug:

Comment (jodh-intel) [reply virtcontainers/kata_shim.go]: Good catch! That chunk should have been removed - branch updated!

**Comment by jodh-intel** [comment]
Metrics detected a couple of issues:

```
16:45:47 time="2018-12-05T16:45:47Z" level=warning msg="Failed Maxval (3.045000 < 3.126150) for [boot-times]"
16:45:47 time="2018-12-05T16:45:47Z" level=warning msg="Check for [boot-times] failed [Failed]"
16:45:47 time="2018-12-05T16:45:47Z" level=warning msg=" with [[*F* boot-times 95.0% 107.8% 105.0% 10.0% 104.0% 120.0% 15.3% 3.1% 20]]"
16:45:47 time="2018-12-05T16:45:47Z" level=warning msg="Failed Minval (137263.525000 > 137061.440000) for [memory-footprint]"
16:45:47 time="2018-12-05T16:45:47Z" level=warning msg="Check for [memory-footprint] failed [Failed]"
16:45:47 time="2018-12-05T16:45:47Z" level=warning msg=" with [[*F* memory-footprint 85.0% 84.9% 115.0% 30.0% 84.9% 84.9% 0.0% 0.0% 1]]"
16:45:47 time="2018-12-05T16:45:47Z" level=warning msg="Overall we failed"
16:45:47 
16:45:47 Report Summary:
16:45:47 +-----+----------------------+-------+--------+--------+-------+--------+--------+-------+------+-----+
16:45:47 | P/F |         NAME         |  FLR  |  MEAN  |  CEIL  |  GAP  |  MIN   |  MAX   |  RNG  | COV  | ITS |
16:45:47 +-----+----------------------+-------+--------+--------+-------+--------+--------+-------+------+-----+
16:45:47 | *F* | boot-times           | 95.0% | 107.8% | 105.0% | 10.0% | 104.0% | 120.0% | 15.3% | 3.1% |  20 |
16:45:47 | *F* | memory-footprint     | 85.0% | 84.9%  | 115.0% | 30.0% | 84.9%  | 84.9%  | 0.0%  | 0.0% |   1 |
16:45:47 | P   | memory-footprint-ksm | 95.0% | 101.2% | 105.0% | 10.0% | 101.2% | 101.2% | 0.0%  | 0.0% |   1 |
16:45:47 +-----+----------------------+-------+--------+--------+-------+--------+--------+-------+------+-----+
```

Re-running as this seems odd given the PR's only adding a couple of bools and an inline-able function...

**Comment by grahamwhaley** [comment]
err, yarp - something looks like it has changed (merged) or broke that has thrown the metrics out. The history on the build machine is very 'red':
http://jenkins.katacontainers.io/computer/x86_packet01/builds

I've not been watching it for the last couple of days. I can have a look tomorrow to see if that was a merge or if the slave has gone bonkers and needs a reset...

**Comment by jodh-intel** [comment]
> The history on the build machine is very 'red':

Well it looks festive atleast! Reminds me of: 🦌🦌🦌🦌🦌🦌🦌🦌

> I can have a look tomorrow

Thanks!

**Comment by jodh-intel** [comment]
Ignoring metrics CI result for now...

**Pull request merged**